cds-snc/cppd-medical-report-sails
# User management

# What this does
Adds a basic User Management functionality, including Create user, Edit user, Delete user.

## To test
- Best to test locally for the moment, because we haven't figured out how to run our seeds on heroku
- Make sure you've got the auth flag enabled in .env (FLAG_AUTH=true)
- Run the database seeders `npm run db:seed` to ensure you have the test users
- Visit `/en/users` if you're not logged in, you should be directed to the login screen
- If you were logged in previously, you'll want to logout, visit: `/en/logout`
- Login using the credentials:
    - username: admin@user.com
    - password: secret
- You should be directed to `/en/users`
- Try creating a user
- Try editing a user (note, you cannot edit the user's password)
- Try deleting a user (note that you can't delete yourself!)

NOTE: The Reset Password link and functionality will be coming in a separate PR

This Feature also includes a full test suite, so try running the cypress tests:
- run `npm run cypress:open`
- Either select Run all specs, or just run the specs under user_managment
